api/desecapi/views.py

```diff
@@ -101,25 +102,41 @@ def get_queryset(self):
         return Domain.objects.filter(owner=self.request.user.pk)
 
     def perform_create(self, serializer):
+        domain_name = serializer.validated_data['name']
+
         pattern = patternDyn if self.request.user.dyn else patternNonDyn
-        if pattern.match(serializer.validated_data['name']) is None:
+        if pattern.match(domain_name) is None:
             ex = ValidationError(detail={
                 "detail": "This domain name is not well-formed, by policy.",
                 "code": "domain-illformed"}
             )
             ex.status_code = status.HTTP_409_CONFLICT
             raise ex
 
-        # Generate a list containing this and all higher-level domain names
-        domain_name = serializer.validated_data['name']
-        domain_parts = domain_name.split('.')
-        domain_list = {'.'.join(domain_parts[i:]) for i in range(1, len(domain_parts))}
-
-        # Remove public suffixes and then use this list to control registration
-        domain_list -= settings.LOCAL_PUBLIC_SUFFIXES
-
-        queryset = Domain.objects.filter(Q(name=domain_name) | (Q(name__in=domain_list) & ~Q(owner=self.request.user)))
-        if queryset.exists():
+        # Check if domain is a public suffix
+        psl = psl_dns.PSL(resolver=settings.PSL_RESOLVER)
+        try:
+            public_suffix = psl.get_public_suffix(domain_name)
+            is_public_suffix = psl.is_public_suffix(domain_name)
+        except psl_dns.exceptions.UnsupportedRule:
+            # Don't be too restrictive if indeterminate
+            public_suffix = domain_name.rsplit('.')[-1]  # `*` PSL rule always applies
```

Comment:
Done. - The `maxsplit` argument is actually of no consequence (unless we set it to `0`).

Reply:
Update: This code went away because I changed error handling such that we reraise the exception in order to produce a 503 along with an email. The view now exits here.